web/src/settings_components.ts

```diff
@@ -1768,3 +1768,107 @@ export function set_custom_time_inputs_visibility(
         $time_select_elem.parent().find(".custom-time-input-container").hide();
     }
 }
+
+type AssignedGroupPermission = {
+    setting_name: RealmGroupSettingName;
+    can_edit: boolean;
+    tooltip_message?: string;
+};
+
+function get_assigned_permission_object(
+    setting_value: GroupSettingValue,
+    setting_name: RealmGroupSettingName,
+    group_id: number,
+    can_edit_settings: boolean,
+): AssignedGroupPermission | undefined {
```

Comment:
Should this helper go in `group_permission_settings.ts`? 

I've not tried to trace its dependencies, but it seems separable from UI code, given its return type.

Reply:
Moved this.